server/src/api/api.ts

```diff
@@ -43,465 +24,572 @@ async function getItem(itemId: number, isAdmin: boolean): Promise<Item|null> {
+            // group by request_item_id
+            // order by request_item_id;
+
+            // the lesser of two evils.  These 2 partial queries give us all the info we need to calculate the two different quantity values
+            const qtyNotAtDesk = await DB.from("requests")
+                .whereIn("status", ["FULFILLED", "LOST", "DAMAGED"])
+                .groupBy("request_item_id")
+                .orderBy("request_item_id")
+                .select("request_item_id")
+                .count("request_id");
+
+            const qtyNotAtDeskObj = {};
+
+            qtyNotAtDesk.forEach((value, index, arr) => {
+                qtyNotAtDeskObj[value.request_item_id.toString()] = parseInt(value.count, 10);
+            });
 
-        return await DB.from("users")
-            .where(searchObj)
-            .select(colNames)
-            .orderBy("name");
-    },
-    /**
-     * Returns information about a single item
-     * TODO: better filtering/search
-     * Access level: any signed in user
-     * @param root
-     * @param _args
-     * @param _context
-     */
-    item: async (root, _args, _context): Promise<Item|null> => {
-        // @ts-ignore
-        const {args, context} = fixArguments(root, _args, _context);
-
-        return await getItem(args.id, context.user.admin);
-    },
-    /**
-     * Bulk items API
-     * TODO: pagination/returned quantity limit
-     * Access level: any signed in user
-     * @param root
-     * @param _args
-     * @param _context
-     */
-    items: async (root, _args, _context): Promise<Item[]> => {
-        // @ts-ignore
-        const {args, context} = fixArguments(root, _args, _context);
-        const items = await DB.from("items")
-            .join("categories", "items.category_id", "=", "categories.category_id");
-
-        return items.map(item => {
-            return {
-                ...item,
-                id: item.item_id,
-                category: item.category_name,
-                price: onlyIfAdmin(item.price, context.user.admin),
-                owner: onlyIfAdmin(item.owner, context.user.admin)
-            };
-        });
-    },
-    categories: async (root, _args, _context): Promise<Category[]> => {
-        // @ts-ignore
-        const {args, context} = fixArguments(root, _args, _context);
-        return await DB.from("categories");
-    },
-    requests: async (root, _args, _context): Promise<Request[]> => {
-        // @ts-ignore
-        const {args, context} = fixArguments(root, _args, _context);
+            console.log(qtyNotAtDeskObj);
 
-        const searchObj: any = {};
+            const qtyPartialReserved = await DB.from("requests")
+                .whereIn("status", ["SUBMITTED", "APPROVED", "READY_FOR_PICKUP"])
+                .groupBy("request_item_id")
+                .orderBy("request_item_id")
+                .select("request_item_id")
+                .count("request_id");
```

Comment:
what does `.count` do?

Reply:
It's the SQL COUNT aggregate operation

Reply:
But I later realized it needs to be SUM on the quantity field